File: Sample/Sample/MessageCenterDelegate.m
/* Copyright Airship and Contributors */

#import "MessageCenterDelegate.h"
#import "MessageCenterViewController.h"
#import "AppDelegate.h"

@interface MessageCenterDelegate ()
@property(nonatomic, strong) UITabBarController *tabBarController;
@property(nonatomic, strong) MessageCenterViewController *messageCenterViewController;
@end

@implementation MessageCenterDelegate

- (instancetype)initWithRootViewController:(UIViewController *)rootViewController {
    self = [super init];
    if (self) {
        self.tabBarController = (UITabBarController *)rootViewController;
        self.messageCenterViewController = [self.tabBarController.viewControllers objectAtIndex:MessageCenterTab];
    }
    return self;
}

- (void)displayMessageCenterAnimated:(BOOL)animated {
    dispatch_async(dispatch_get_main_queue(), ^{
        self.tabBarController.selectedIndex = MessageCenterTab;
        [self.messageCenterViewController showInbox];
    });
}

- (void)displayMessageCenterForMessageID:(NSString *)messageID animated:(BOOL)animated {
    [self displayMessageCenterAnimated:animated];
    dispatch_async(dispatch_get_main_queue(), ^{
        [self.messageCenterViewController displayMessageForID:messageID];
    });
}

 

- (void)dismissMessageCenterAnimated:(BOOL)animated {
    // no-op
}


@end

File: Sample/Sample/MessageCenterDelegate.swift
/* Copyright Airship and Contributors */

import Foundation
import AirshipCore
import AirshipMessageCenter

class MessageCenterDelegate : NSObject, MessageCenterDisplayDelegate {
    var tabBarController : UITabBarController;
    var messageCenterViewController : MessageCenterViewController;

    init(tabBarController:UITabBarController) {
        self.tabBarController = tabBarController
        let appDelegate = UIApplication.shared.delegate as! AppDelegate
        self.messageCenterViewController = self.tabBarController.viewControllers![appDelegate.MessageCenterTab] as! MessageCenterViewController;
    }

    func displayMessageCenter(forMessageID messageID: String, animated: Bool) {
        self.displayMessageCenter(animated: animated)
        DispatchQueue.main.async {
            self.messageCenterViewController.displayMessage(forID: messageID)
        }
    }

    func displayMessageCenter(animated: Bool) {
        DispatchQueue.main.async {
            let appDelegate = UIApplication.shared.delegate as! AppDelegate
            self.tabBarController.selectedIndex = appDelegate.MessageCenterTab
            self.messageCenterViewController.showInbox()
        }
    }

    func dismissMessageCenter(animated: Bool) {
        // no-op
    }
}


